crates/forge/src/running.rs

```diff
@@ -70,25 +71,20 @@ fn build_hints_dict<'b>(
     (hints_dict, string_to_hint)
 }
 
-pub(crate) async fn blocking_run_from_test(
+pub(crate) fn blocking_run_from_test(
     args: Vec<Felt252>,
     case: Arc<TestCase>,
     runner: Arc<SierraCasmRunner>,
     runner_config: Arc<RunnerConfig>,
     runner_params: Arc<RunnerParams>,
-    sender: Option<Sender<()>>,
-) -> Result<TestCaseSummary> {
+    send: Sender<()>,
+) -> JoinHandle<Result<TestCaseSummary>> {
     tokio::task::spawn_blocking(move || {
-        run_test_case(
-            args,
-            &case,
-            &runner,
-            &runner_config,
-            &runner_params,
-            &sender,
-        )
+        if send.is_closed() {
```

Comment:
What if the channel is closed after we are past this section?

Reply:
I think we'll get an error like previously. Perhaps this should be approached the other way round - "main" task should wait for all blocking tasks to terminate? Channel or some other messaging could be used for that (or maybe a simpler solution).

Reply:
in most cases, this test will be executed but user will not get the result/error. I think in extreme situations we will get >= x error like the previous, where x is the number of user cores.

Reply:
@Arcticae @cptartur  https://github.com/foundry-rs/starknet-foundry/pull/874/commits/792532dd89e4eaa10bc6ba5c4d67cea9a7bc7a2b

I fixed this problem

Reply:
I think this should resolve the issue. We should add a comment somewhere in the code (probably in `run()`) explaining the logic of using two channels for this.

Reply:
👍